### ISSUE TITLE
Document how to navigate between blank page and navigation pane.

### ISSUE THREAD

**Issue by Tostifrosti2** [Microsoft/WindowsTemplateStudio#775]
When you create a new project with Windows Template Studios, you'll get a wizard. In this wizard, you can choose between 'Navigation Pane' and 'Blank Page'. Why isn't it possible to merge those options together?

In real world scenario, you may want to start with a login page and after logging in, you will be navigated to the 'Navigate Pane' page.

**Comment by mrlacey** [comment]
@Tostifrosti2 We already have plans for introducing the ability to add a login page as the first page seen in issue #291 

The **Blank** project type is intended for if you want to completely control the structure and in-app navigation yourself. Your request for a

Until this is built in I expect it'll be beneficial to add some docs to explain how to do this yourself.  
I'm going to change this issue to be about adding the documentation and it should get done soon.

**Comment by Tostifrosti2** [comment]
@mrlacey Thank you for the quick response!

I have already started a project where I used the MVVM Light as start-up project.
I noticed that the NavigationService is used to switch between pages using the Window.Current.Content as Frame (that was initialised in the ActivationService). 

I've tried to create a method myself to set the frame, but with no success.

`public bool NavigateToFrame(string pageKey, UIElement frame, object parameter = null, NavigationTransitionInfo infoOverride = null)
        {
            lock (_pages)
            {
                if (!_pages.ContainsKey(pageKey))
                {
                    throw new ArgumentException($"Page not found: {pageKey}. Did you forget to call NavigationService.Configure?", "pagekey");
                }
                if (frame != null)
                {
                    Frame.BackStack.Clear();
                    Window.Current.Content = frame;
                    _frame = null;
                }
                var navigationResult = Frame.Navigate(_pages[pageKey], parameter, infoOverride);
                return navigationResult;
            }
        }`

This method above will result in a NullpointerException, because the Frame is null. The weird thing is, is if I debug this code above, the application just runs fine. :S
Could you help me out?

**Comment by Tostifrosti2** [comment]
@mrlacey Thank you for your help, but i've already solved the problem.

I've changed the method a bit and overridden the 'OnNavigatedTo' method in the MainPage (Navigation Page).
In this 'OnNavigatedTo' i give the Frame to the ViewModel so the ViewModel can initialize the Frame in the NavigationService. 

NavigationService.cs
```
public bool NavigateToFrame(string pageKey, UIElement frame, object parameter = null, NavigationTransitionInfo infoOverride = null)
        {
            lock (_pages)
            {
                if (!_pages.ContainsKey(pageKey))
                {
                    throw new ArgumentException($"Page not found: {pageKey}. Did you forget to call NavigationService.Configure?", "pagekey");
                }
                if (frame != null)
                {
                    RemoveHistory();
                    Window.Current.Content = frame;
                    Window.Current.Activate();
                }
                var navigationResult = Frame.Navigate(_pages[pageKey], parameter, infoOverride);
                return navigationResult;
            }
        }
```
MainPage.xaml.cs
```
protected override void OnNavigatedTo(NavigationEventArgs e)
        {
            base.OnNavigatedTo(e);
            ViewModel.Initialize(MainFrame);
        }
```
MainPageViewModel.cs
```
public void Initialize(Frame frame)
        {
            NavigationService.Frame = frame;
            NavigationService.Frame.Navigated += NavigationService_Navigated;
            PopulateNavItems();
        }
```

**Comment by mrlacey** [comment]
@Tostifrosti2 glad you found a solution. I still think it's worth creating the docs to help other though.

**Comment by Tostifrosti2** [comment]
PR created: https://github.com/Microsoft/WindowsTemplateStudio/pull/783

**Comment by mrlacey** [comment]
Notes to self.

Docs must:
- Cover Code Behind, MVVMBasic, and MVVM Light
- Not break any existing other features that use activation handlers

May be related to #697

**Comment by mvegaca** [comment]
I'm goint to work on a documentation for navigation service and activation

**Comment by mvegaca** [comment]
Working on NavigationService updates and fixes on #1012

**Comment by ralarcon** [comment]
Closing. We currently have the Navigation [sample](https://github.com/Microsoft/WindowsTemplateStudio/tree/dev/samples/navigation/MixedNavigationSample) to showcase this.

**Comment by mrlacey** [comment]
I don't believe this is ready to close out.
- The docs to explain this basically just say look at the sample. With only a very basic explanation of the differences made to the sample project to demonstrate the change.
- The sample project does not compile which prevents people from trying it for themselves. (missing pfx file)
- The sample project does not have the changes made to it clearly documented. Doing this would help highlight what was changed and make the sample show the changes more easily to people not familiar with the generated codebase.
- The sample is dependent on changes in the generated code added for version 1.3. This should be called out for anyone trying to add this functionality to a project generated with an earlier version.
- The sample is only available for MVVM Basic. Doing the same in MVVM Light has some important differences which should also be documented here. A description of what's different if using Code Behind should also be included.
- One of the key drivers for having this documentation is for the scenario around having a login page before accessing the main content of the app. Connected with this is the ability to also log out. Upon logging out the app must return to the original frame and not show pages within the shell. Without this, the documentation is only half complete.

**Comment by crutkas** [comment]
@mvegaca, did you see the community PR doc for this?  https://github.com/Microsoft/WindowsTemplateStudio/pull/783/files

**Comment by mvegaca** [comment]
Hi, Matt.
Done documentation and sample updates on #1063.

Documentation updates:
- Added a new section to explain navigation concepts between different project types.
- Added some explanations on sample project and references in doc.
- Added details on sample project configuration (framework and WTS version).

**Comment by mvegaca** [comment]
[Getting started endusers](https://github.com/mvegaca/WindowsTemplateStudio/blob/navigation-sample/docs/getting-started-endusers.md) documentation updated on PR

**Comment by mvegaca** [comment]
In relation to #782
- [Navigation documentation](https://github.com/Microsoft/WindowsTemplateStudio/blob/dev/docs/navigation.md) updates are already covering how generated code handle navigation between pages on different project types and frameworks.
- [MixedNavigationSample](https://github.com/Microsoft/WindowsTemplateStudio/tree/dev/samples/navigation/MixedNavigationSample) covers the described scenario in a generated App.

![image](https://user-images.githubusercontent.com/5556784/30056691-f96e2de8-9233-11e7-83df-24324d4933c3.png)

I'm not sure that if necessary to create the sample app on three different frameworks because the differences between frameworks are minimal and there are explained in the documentation.

**Comment by mvegaca** [comment]
I think it can be closed

**Comment by mrlacey** [comment]
After spending too much time looking at it, I couldn't work out how to get an MVVM Light version of the sample to work. 
It requires extra registration steps which the other frameworks don't. It also isn't obvious how the ShellViewModel should be registered. Something that isn't needed otherwise. (Registering it like another page doesn't work.)
This needs documenting.

**Comment by mvegaca** [comment]
Pull request #1092

- MixedNavigationSample created for three different frameworks.
- Documentation updated.

**Comment by mvegaca** [comment]
Can it be closed?

**Comment by ralarcon** [comment]
A native english speaker should have a look to the documentation :-D. I think is good enough by the moment so as someone review it, we can close this issue.

**Comment by ralarcon** [comment]
@crutkas, @mrlacey can you have a look to the document and if it is good enough, close the issue, if not, remove the "Can close out soon" and move it to 1.4. Thanks!

**Comment by mrlacey** [comment]
> One of the key drivers for having this documentation is for the scenario around having a login page before accessing the main content of the app. Connected with this is the ability to also log out. Upon logging out the app must return to the original frame and not show pages within the shell. Without this, the documentation is only half complete.

This [from the other day](https://github.com/Microsoft/WindowsTemplateStudio/issues/775#issuecomment-327033161) has not been addressed and so I say punt it to the next version.

How to do the above isn't obvious to me based on the NavigationService, even with the [recent changes](https://github.com/Microsoft/WindowsTemplateStudio/pull/1016). 

The other question related to this is: How can a page in an app using NavPanel layout show a page without the Hamburger menu being visible? Such a request is quite common for many apps I've worked on over the years, to create a "full-screen" experience but leave the back navigation button visible.

How each of these should be addressed will likely be different for different project types and framework combinations.

**However**, rather than just let the new samples sit without any documentation I've made some changes and submitted PR #1109.
If this PR is accepted and this issue closed, a follow-up issue should be created track creating documentation to cover the above-identified scenarios.

**Comment by ralarcon** [comment]
@mrlacey, these things you point can be part of a second round of improvements. To have a feature which includes the login / log-out capabilities would be great too so we can extend and improve the documentation aligned with the development of this new feature. 

The "Full-Screen" view looks to me as a new capability for the NavigatioService... really interesting, by the way.

We are going to close this and open a new issue to track the Navigation service improvements and their documentation and x-ref to this.

**Comment by crutkas** [comment]
Did this get resolved?  I want to close out 1.3

**Comment by mrlacey** [comment]
> Did this get resolved?

Follow-up work was not done nor an issue raised to carry it out separately. One of these must be done before it can be closed.

**Comment by ralarcon** [comment]
Added #1171 to track navigation sample and doc improvements.

**Comment by ralarcon** [comment]
Closing now as this is done.
